src/System.Net.Sockets/src/System/Net/Sockets/SocketPal.Unix.cs

```diff
@@ -718,6 +718,7 @@ public static bool TryCompleteSendTo(SafeCloseSocket socket, ReadOnlySpan<byte>
             {
                 int sent;
                 Interop.Error errno;
+                bool successfulSend = false;
```

Comment:
It seems like this should be outside the for loop, otherwise you're resetting it on every loop.  Am I missing something here?

Reply:
That does seem wrong (and probably explains why the test is failing).

Reply:
Yes, this must be higher up.